pkg/scheduler/objects/application_summary.go

```diff
@@ -0,0 +1,65 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package objects
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/yunikorn-core/pkg/common/resources"
+	"github.com/apache/yunikorn-core/pkg/log"
+	"go.uber.org/zap"
+)
+
+type ApplicationSummary struct {
+	ApplicationID       string
+	SubmissionTime      time.Time
+	StartTime           time.Time
+	FinishTime          time.Time
+	User                string
+	Queue               string
+	State               string
+	RmID                string
+	ResourceUsage       *resources.TrackedResource
+	PreemptedResource   *resources.TrackedResource
+	PlaceholderResource *resources.TrackedResource
+}
+
+func (as *ApplicationSummary) String() string {
+	if as == nil {
+		return "nil ApplicationSummary"
+	}
```

Comment:
Do we need this "nil" check?

Reply:
For now, we don't need this nil check. I'll remove the nil check and test code in the next commit, many thanks.